homeassistant/components/mqtt/quality_scale.yaml

```diff
@@ -0,0 +1,122 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup:
+    status: done
+    comment: >
+      Entities are updated through dispatchers, and these are
+      cleaned up when the integration unloads.
+  entity-unique-id:
+    status: exempt
+    comment: >
+      This is user configurable, but not required.
+      It is required though when a user wants to use device based discovery.
+  has-entity-name: done
+  runtime-data:
+    status: exempt
+    comment: >
+      Runtime data is not used, as the mqtt entry data is only used to set up the
+      MQTT broker, this happens during integration setup,
+      and only one config entry is allowed.
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
```

Comment:
Where is this rule fulfilled? Is this somewhere atypical or should this be exempt if it doesn't have a uique id?

Reply:
https://github.com/home-assistant/core/pull/131448 ensures a unique id is assigned.

Reply:
This rule isn't about forcing the unique ID. It is a common way, but the gist is to ensure each entry is unique, which is guarded already.

Reply:
Agreed, it is not necessary to updated to set unique id given the use of `single_config_entry`